__tests__/allTests.test.js

```diff
@@ -0,0 +1,233 @@
+      expect(body).toHaveProperty('GET /api/articles');
+      expect(body).toHaveProperty('GET /api/articles/:article_id');
+      expect(body).toHaveProperty('GET /api/topics');
+  
+      expect(body['GET /api']).toMatchObject({
+        description: expect.any(String),
+      });
+  
+      expect(body['GET /api/articles']).toMatchObject({
+        description: expect.any(String),
+        exampleResponse: expect.any(Object),
+        queries: expect.any(Array),
+      });
+
+      //console.log(body['GET /api/articles/:article_id'])
+  
+      expect(body['GET /api/articles/:article_id']).toMatchObject({
+        article_id: expect.any(String),
+        article_img_url: expect.any(String),
+        author: expect.any(String),
+        created_at: expect.any(String),
+        title: expect.any(String),
+        topic: expect.any(String),
+        votes: expect.any(Number),
+      });
+  
+        //console.log(body['GET /api/topics'])
+
+      expect(body['GET /api/topics']).toMatchObject({
+        description: expect.any(String),
+        exampleResponse: expect.any(Object),
+        queries: expect.any(Array),
+      });
+  
+      expect(body['GET /api/topics'].exampleResponse.topics).toBeInstanceOf(Array);
+      body['GET /api/topics'].exampleResponse.topics.forEach((topic) => {
+        expect(topic).toMatchObject({
+          slug: expect.any(String),
+          description: expect.any(String),
+        });
+      });
+    });
+  }
+);
+
+
+// fix this later - doesnt work
+describe('/api/articles/:articles_id', () => {
+    test('GET:200 sends an object of topics to the client when making a valid request', () => {
+        return request(app)
+        .get('/api/articles/')
+        .expect(200)
+        .then(({body}) =>{
+            let obj = body[0]
+          //  console.log(obj)
+
+            expect(obj).toMatchObject({
+              article_id: expect.any(Number),
+              title: expect.any(String),
+              topic: expect.any(String),
+              created_at: expect.any(String),
+              votes: expect.any(Number),
+              article_img_url: expect.any(String),
+              comment_count: expect.any(Number)
+            })
+            //expect(body.topics).toHaveLength(3)
+      
+
+        });
+    })
+
+
+    test('GET:200 sends undefined to the client when making an invalid number request for the articles', () => {
+      return request(app)
+      .get('/api/articles/9999')
+      .expect(200)
+      .then(({body}) =>{
+          let obj = {body}
+
+          let obj2 = obj[0];
+
+        //  console.log(obj2)
+
+         expect(obj2).toBe(undefined)
+      });
+  })
+})
+
+
+
+
+describe('/api/articles/', () => {
+  test('GET:200 sends an object of topics to the client when making a valid request', () => {
+      return request(app)
+      .get('/api/articles/1')
+      .expect(200)
+      .then(({body}) =>{
+          let obj = body
+
+          let obj2 = obj[0]
```

Comment:
obj is a generic name that doesn't really give you much info about what the variable is - I'd rename this to what it actually is, like article. You'd always want descriptive variable names